### DISCUSSION TITLE
Extending user model with File filed like user avatar

### DISCUSSION THREAD

**Discussion by Tur8008** [fastapi-users/fastapi-users#1253] [Q&A] [answered]
Hello.

I need to add user avatar as a part of user's profile. And it seems like I can't use `json` here to pass with request to `/users/me` or `users/:id endpoint. That's because FastAPI demands using form for file uploading.

I see now that I have to override `update_me` and update_user routers so that they take form instead of json schema?
 Actually I'm not going to store images in a db. I just  need to store the path to the storage.

Is there the better way to implement it?

P.S. Thank you a lot for your lib! It makes me happy!

**Comment by frankie567** [comment]
For specific things like image uploads, I would recommend to implement a dedicated endpoint for that. Typically, it would accept form data in the request, which you can then process to validate it's an image, crop/resize it, store it in a cloud storage, etc.

Here is a snippet from one of my project:

```py
@router.post("/users/me/profile-picture", response_model=UserRead)
async def upload_profile_picture(
    file: UploadFile = File(...),
    user: User = Depends(current_active_user),
    storage: Storage = Depends(get_storage),
):
    content_type = file.content_type

    if not content_type.startswith("image/"):
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=ErrorCode.USER_INVALID_PROFILE_PICTURE_IMAGE,
        )

    picture_key = f"{user.id}.profile_picture"
    temp_file = cast(tempfile.SpooledTemporaryFile, file.file)
    storage.upload_profile_picture(picture_key, temp_file, content_type)

    user = await manager.user_db.update(
        user, {"profile_picture_storage_key": picture_key}
    )

    return user

```

**Reply by Tur8008** [reply]
I see your point, thanks. The only thing I'm not sure is the case if I have a form with other fields such `nickname`, `first name` ... and so on I have to send at least two different requests and then displaying the results separately on the form.

**Reply by frankie567** [reply, marked as answer]
Indeed, but from what I've seen in the wild, lot of web apps nowadays have a separate form for profile picture (like GitHub for example).
